src/k8s_lb_manager.py

```diff
@@ -0,0 +1,237 @@
+# - Allows alphanumeric characters, dots (.), dashes (-), and underscores (_)
+# - Matches the entire string
+# - Does not allow empty strings
+# - Example valid: "value1", "my-value", "value.name", "value_name"
+# - Example invalid: "value@", "value#", "value space"
+ANNOTATION_VALUE_PATTERN = re.compile(r"^[\w.\-_]+$")
+
+# Based on https://github.com/kubernetes/apimachinery/blob/v0.31.3/pkg/util/validation/validation.go#L204
+# Regex for DNS1123 subdomains:
+# - Starts with a lowercase letter or number ([a-z0-9])
+# - May contain dashes (-), but not consecutively, and must not start or end with them
+# - Segments can be separated by dots (.)
+# - Example valid: "example.com", "my-app.io", "sub.domain"
+# - Example invalid: "-example.com", "example..com", "example-.com"
+DNS1123_SUBDOMAIN_PATTERN = re.compile(
+    r"^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+)
+
+# Based on https://github.com/kubernetes/apimachinery/blob/v0.31.3/pkg/util/validation/validation.go#L32
+# Regex for Kubernetes qualified names:
+# - Starts with an alphanumeric character ([A-Za-z0-9])
+# - Can include dashes (-), underscores (_), dots (.), or alphanumeric characters in the middle
+# - Ends with an alphanumeric character
+# - Must not be empty
+# - Example valid: "annotation", "my.annotation", "annotation-name"
+# - Example invalid: ".annotation", "annotation.", "-annotation", "annotation@key"
+QUALIFIED_NAME_PATTERN = re.compile(r"^[A-Za-z0-9]([-A-Za-z0-9_.]*[A-Za-z0-9])?$")
+
+
+class KubernetesLoadBalancer:
+    """KubernetesLoadBalancer."""
+
+    def __init__(
+        self,
+        name: str,
+        namespace: str,
+        field_manager: str,
+        ports: List[ServicePort],
+        additional_labels: Optional[Dict[str, str]] = None,
+        additional_selectors: Optional[Dict[str, str]] = None,
+        additional_annotations: Optional[str] = None,
+    ):
+        """Initialize the KubernetesLoadBalancer.
+
+        :param name: Name of the LoadBalancer.
+        :param namespace: Namespace for the LoadBalancer.
+        :param ports: List of ServicePort objects.
+        :param additional_labels: Additional labels to apply.
+        :param additional_selectors: Additional selectors to match pods.
+        :param additional_annotations: Additional annotations to apply.
+        """
+        self.name = name
+        self.namespace = namespace
+        self.ports = ports
+        self.field_manager = field_manager
+        self.additional_labels = additional_labels or {}
+        self.additional_selectors = additional_selectors or {}
+        self.additional_annotations = additional_annotations or {}
+        self.additional_annotations = parse_annotations(additional_annotations)
+        # Initialize Kubernetes client
+        self.client = Client(namespace=self.namespace, field_manager=self.field_manager)
+
+        if self._annotations_valid():
+            self.reconcile()
+        else:
+            self.remove_lb()
```

Comment:
in particular, `self.remove_lb()` feels like an unexpected thing to call in a constructor